app/components/NavMenu/index.tsx

```diff
@@ -96,12 +97,32 @@ interface Props {
 export const NavMenu: FC<Props> = (props) => {
   const locale = useSelector(selectLocale);
   const balances = useSelector(selectBalances);
-  const { pathname } = useLocation();
+  const domains = useSelector(selectDomain);
+  const [domain, setDomain] = useState<Domain | undefined>(undefined);
+  const svg = useRef(null);
+  useEffect(() => {
+    if (domains?.knsDomain.name) {
+      setDomain(domains.knsDomain);
+    } else if (domains?.ensDomain.name) {
+      setDomain(domains.ensDomain);
+    }
+  }, [domains]);
+  useEffect(() => {
+    if (svg.current && domain && domain.defaultImage) {
+      (svg.current as any).appendChild(htmlToElement(domain.defaultImage));
+    }
+  }, [domain]);
 
+  const { pathname } = useLocation();
   const handleHide = () => {
     props.onHide?.();
   };
-
+  const htmlToElement = (htmlString: string): Element => {
+    const template = document.createElement("template");
+    htmlString = htmlString.trim(); // Never return a text node of whitespace as the result
+    template.innerHTML = htmlString;
+    return template.content.firstChild as Element;
+  };
```

Comment:
did you try this:

```jsx
<div dangerouslySetInnerHTML={{ __html: domain.defaultImage }} />
```

It should be safe enough because
1. it only works if it's the user's own kns/ens domain via their connected wallet
2. its the default image, not user generated

Reply:
cool, yea good point. updated.